src/systems/filecoin_mining/storage_mining/storage_miner_actor.id

```diff
@@ -1,166 +1,99 @@
-    CommittedSectors   sector.CompactSectorSet
-    RecoveringSectors  sector.CompactSectorSet
-    FailingSectors     sector.CompactSectorSet
-
-    // transient State that get reset on every constructPowerReport
-    TerminatedFaults   sector.CompactSectorSet
+type SectorState enum {
+    PreCommit
+    Active
+    TemporaryFault
 }
 
 type SectorOnChainInfo struct {
-    SealCommitment  sector.SealCommitment
-    State           SectorState
-    SectorWeight    block.SectorWeight
-    Activation      block.ChainEpoch
-    Expiration      block.ChainEpoch
-}
-
-type ChallengeStatus struct {
-    LastChallengeEpoch     block.ChainEpoch  // get updated by NotifyOfPoStSurpriseChallenge and SubmitElectionPoSt
-    _lastPoStFailureEpoch  block.ChainEpoch  // get updated upon _onMissedSurprisePoSt
-    _lastPoStSuccessEpoch  block.ChainEpoch  // get updated by successful Election or Suprise PoSt submission
-
-    _getStoragePowerActorState(stateTree stateTree.StateTree) spc.StoragePowerActorState
-
-    OnNewChallenge(currEpoch block.ChainEpoch)
-    LastPoStResponseEpoch() block.ChainEpoch
-    LastPoStSuccessEpoch() block.ChainEpoch
-    OnPoStSuccess(currEpoch block.ChainEpoch)
-    OnPoStFailure(currEpoch block.ChainEpoch)
-
-    IsChallenged() bool  // only True when proving SurprisePoSt (implicit because ElectionPoSt completes within a block)
-    ChallengeHasExpired(currEpoch block.ChainEpoch) bool
-    CanBeElected(currEpoch block.ChainEpoch) bool
-    ShouldChallenge(
-        currEpoch block.ChainEpoch
-    ) bool
-}
-
-type PreCommittedSector struct {
-    Info           sealing.SectorPreCommitInfo
-    ReceivedEpoch  block.ChainEpoch
+    State                       SectorState
+    Info                        sealing.SectorPreCommitInfo  // Also contains Expiration field.
+    PreCommitDeposit            actor.TokenAmount
+    PreCommitEpoch              block.ChainEpoch
+    ActivationEpoch             block.ChainEpoch  // -1 if still in PreCommit state.
+    DeclaredFaultEpoch          block.ChainEpoch  // -1 if not currently declared faulted.
+    DeclaredFaultDuration       block.ChainEpoch  // -1 if not currently declared faulted.
+    DealWeight                  BigInt  // -1 if not yet validated with StorageMarketActor.
+
+    // Must be significantly larger than DeclaredFaultEpoch, since otherwise it may be possible
+    // to declare faults adaptively in order to exempt challenged sectors.
+    EffectiveFaultBeginEpoch()  block.ChainEpoch
+    EffectiveFaultEndEpoch()    block.ChainEpoch
+
+    Is_TemporaryFault()         bool
 }
 
-type PreCommittedSectorsAMT {sector.SectorNumber: PreCommittedSector}
-type StagedCommittedSectorAMT {sector.SectorNumber: SectorOnChainInfo}
 type SectorsAMT {sector.SectorNumber: SectorOnChainInfo}
 
-// Balance of a StorageMinerActor should equal exactly the sum of PreCommit deposits that are not yet returned or burned
+type SectorNumberSetHAMT {sector.SectorNumber: bool}
```

Comment:
I think we developed the RLE+ encoding specifically to represent sets of sector numbers efficiently.

Reply:
This is currently used for the ProvingSet, and it's not clear RLE+ gives a more efficient encoding in that case (especially as sector numbers continue to be incremented over time).